File: stepik-java-api/src/main/java/org/stepik/api/queries/views/StepikViewsPostQuery.java
package org.stepik.api.queries.views;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.views.Views;
import org.stepik.api.objects.views.ViewsPost;
import org.stepik.api.queries.StepikAbstractPostQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikViewsPostQuery extends StepikAbstractPostQuery<Views> {
    private final ViewsPost views = new ViewsPost();

    public StepikViewsPostQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Views.class);
    }

    @NotNull
    public StepikViewsPostQuery step(long id) {
        views.getView().setStep(id);
        return this;
    }

    @NotNull
    public StepikViewsPostQuery assignment(@Nullable Long assignment) {
        views.getView().setAssignment(assignment);
        return this;
    }

    @NotNull
    @Override
    protected String getBody() {
        return getJsonConverter().toJson(views, false);
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.VIEWS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/views/StepikViewsPostQuery.kt
package org.stepik.api.queries.views

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.views.Views
import org.stepik.api.objects.views.ViewsPost
import org.stepik.api.queries.StepikAbstractPostQuery

class StepikViewsPostQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractPostQuery<Views>(stepikAction, Views::class.java) {
    
    private val views = ViewsPost()
    
    fun step(id: Long): StepikViewsPostQuery {
        views.view.step = id
        return this
    }
    
    fun assignment(assignment: Long?): StepikViewsPostQuery {
        views.view.assignment = assignment
        return this
    }
    
    override fun getBody(): String {
        return jsonConverter.toJson(views, false)
    }
    
    override fun getUrl(): String {
        return stepikAction.stepikApiClient.host + "/api/views"
    }
}
